src/update-changelog.ts

```diff
@@ -250,19 +250,16 @@ export async function updateChangelog({
   // Ensure release header exists, if necessary
   if (
     isReleaseCandidate &&
+    currentVersion &&
     !changelog
       .getReleases()
       .find((release) => release.version === currentVersion)
   ) {
-    // Typecast: currentVersion will be defined here due to type guard at the
-    // top of this function.
-    changelog.addRelease({ version: currentVersion as Version });
+    changelog.addRelease({ version: currentVersion });
```

Comment:
This comment is referring to this code at the top of the function:

``` typescript
if (isReleaseCandidate && !currentVersion) {
  throw new Error(
    `A version must be specified if 'isReleaseCandidate' is set.`,
  );
}
```

Since we're checking for `isReleaseCandidate` here, there's no way that `currentVersion` is not set. Hence why I believe that the typecast was added originally. (Why TypeScript doesn't see this, I'm not sure.)

This change doesn't hurt, but I wanted to make sure you saw that.